app/controllers/v1/base_controller.rb

```diff
@@ -11,17 +14,30 @@ def rescue_error_messages
     end
   end
 
-  rescue_from ActionController::ParameterMissing do |exception|
-    render json: { error: "#{exception.param} is required" }, status: 422
+  def missing_parameter(error)
+    render json: { error: "#{error.param} is required" }, status: 422
+  end
+
+  def record_not_found(error)
+    render json: {error: error.message}, status: :not_found
+  end
+
+  def default_person_params
+    params.require(:person).permit(Person::ALL_AVAILABLE_ATTRIBUTES)
+  end
+
+  def person_from_params
```

Comment:
hmmm... I see the benefit to this, as far as DRYing up the code goes, but I'm not a fan of using BaseController as grab bag of publicly available methods. Makes it harder to parse what's happening on the controllers that call these methods, and BaseController could end up getting pretty messy